## Creating a Logistic Regression to Predict Absenteeism

**Import the relevant libraries**

In [1]:
import pandas as pd
import numpy as np

**Load the data**

In [2]:
data_preprocessed = pd.read_csv(r'C:\Users\MOJI\Desktop\PortfolioP\Absenteeism_preprocessed_data.csv')
data_preprocessed.head()

Reason_1  Reason_2  Reason_3  Reason_4  Month Value  Day of the Week  \
0         0         0         0         1            7                1   
1         0         0         0         0            7                1   
2         0         0         0         1            7                2   
3         1         0         0         0            7                3   
4         0         0         0         1            7                3   

   Transportation Expense  Distance to Work  Age  Daily Work Load Average  \
0                     289                36   33                  239.554   
1                     118                13   50                  239.554   
2                     179                51   38                  239.554   
3                     279                 5   39                  239.554   
4                     289                36   33                  239.554   

   Body Mass Index  Education  Children  Pets  Absenteeism Time in Hours  
0               30          0         2     1                          4  
1               31          0         1     0                          0  
2               31          0         0     0                          2  
3               24          0         2     0                          4  
4               30          0         2     1                          2

**Create Targets**

In [3]:
# we will take the median value of the 'Absenteeism Time in Hours' and use it as a cut-off line
data_preprocessed['Absenteeism Time in Hours'].median()

3.0

NB: <= 3 hours will be moderately absent, >= 4 hours will Excessively absent

In [4]:
targets = np.where(data_preprocessed['Absenteeism Time in Hours'] > 
                  data_preprocessed['Absenteeism Time in Hours'].median(), 1, 0)

In [5]:
targets

array([1, 0, 0, 1, 0, 0, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 0, 1, 1, 0, 1, 0,
       1, 1, 1, 1, 0, 1, 1, 0, 1, 1, 1, 1, 0, 0, 1, 1, 1, 1, 0, 1, 1, 1,
       0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 1, 1, 0, 1, 1, 0, 1, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1,
       0, 1, 0, 1, 0, 1, 0, 0, 1, 0, 0, 1, 0, 1, 0, 1, 1, 1, 0, 0, 0, 1,
       0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0,
       0, 0, 0, 0, 0, 0, 1, 1, 0, 1, 0, 1, 1, 1, 0, 0, 0, 1, 0, 1, 0, 0,
       1, 0, 1, 0, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 0, 1,
       0, 1, 1, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1, 1, 0, 0, 1, 0, 1, 1,
       1, 1, 1, 1, 0, 0, 1, 0, 0, 1, 1, 1, 0, 0, 1, 0, 0, 0, 0, 0, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 0, 0, 0, 1, 1, 1, 1, 1, 1,
       0, 1, 1, 0, 1, 1, 1, 1, 1, 0, 1, 0, 0, 1, 0, 0, 1, 1, 0, 1, 0, 0,
       0, 0, 0, 1, 1, 1, 1, 0, 1, 0, 1, 0, 0, 0, 1, 1, 0, 1, 1, 1, 0, 0,
       0, 1, 1, 1, 1, 1, 1, 0, 0, 1, 0, 1, 0, 1, 0,

In [6]:
data_preprocessed['Excessive Absenteeism'] = targets

In [7]:
data_preprocessed.head()

Reason_1  Reason_2  Reason_3  Reason_4  Month Value  Day of the Week  \
0         0         0         0         1            7                1   
1         0         0         0         0            7                1   
2         0         0         0         1            7                2   
3         1         0         0         0            7                3   
4         0         0         0         1            7                3   

   Transportation Expense  Distance to Work  Age  Daily Work Load Average  \
0                     289                36   33                  239.554   
1                     118                13   50                  239.554   
2                     179                51   38                  239.554   
3                     279                 5   39                  239.554   
4                     289                36   33                  239.554   

   Body Mass Index  Education  Children  Pets  Absenteeism Time in Hours  \
0               30          0         2     1                          4   
1               31          0         1     0                          0   
2               31          0         0     0                          2   
3               24          0         2     0                          4   
4               30          0         2     1                          2   

   Excessive Absenteeism  
0                      1  
1                      0  
2                      0  
3                      1  
4                      0

Using the median as a cut-off line is numerically stable and rigid, and this is because by using median we have implicitly balanced the dataset, roughly half of it to zero and the other half ones. This will prevent our model from learning to output only 0s or only 1s

**A comment on the targets**

In [8]:
targets.sum() / targets.shape[0]

0.45571428571428574

This result shows that around 46% of the targets are 1s and 56% are 0s. For logistic regression, this is fairly ok

**Drop Absenteeism Time in Hours**

In [9]:
data_with_targets = data_preprocessed.drop(['Absenteeism Time in Hours', 'Day of the Week',
                                            'Daily Work Load Average', 'Distance to Work'], axis = 1)

In [10]:
data_with_targets is data_preprocessed

False

In [11]:
data_with_targets.head()

Reason_1  Reason_2  Reason_3  Reason_4  Month Value  \
0         0         0         0         1            7   
1         0         0         0         0            7   
2         0         0         0         1            7   
3         1         0         0         0            7   
4         0         0         0         1            7   

   Transportation Expense  Age  Body Mass Index  Education  Children  Pets  \
0                     289   33               30          0         2     1   
1                     118   50               31          0         1     0   
2                     179   38               31          0         0     0   
3                     279   39               24          0         2     0   
4                     289   33               30          0         2     1   

   Excessive Absenteeism  
0                      1  
1                      0  
2                      0  
3                      1  
4                      0

### Select the Inputs for the regression

In [12]:
data_with_targets.shape

(700, 12)

In [13]:
# data_with_targets.iloc[:,:14]
data_with_targets.iloc[:,:-1]

Reason_1  Reason_2  Reason_3  Reason_4  Month Value  \
0           0         0         0         1            7   
1           0         0         0         0            7   
2           0         0         0         1            7   
3           1         0         0         0            7   
4           0         0         0         1            7   
..        ...       ...       ...       ...          ...   
695         1         0         0         0            5   
696         1         0         0         0            5   
697         1         0         0         0            5   
698         0         0         0         1            5   
699         0         0         0         1            5   

     Transportation Expense  Age  Body Mass Index  Education  Children  Pets  
0                       289   33               30          0         2     1  
1                       118   50               31          0         1     0  
2                       179   38               31          0         0     0  
3                       279   39               24          0         2     0  
4                       289   33               30          0         2     1  
..                      ...  ...              ...        ...       ...   ...  
695                     179   40               22          1         2     0  
696                     225   28               24          0         1     2  
697                     330   28               25          1         0     0  
698                     235   32               25          1         0     0  
699                     291   40               25          0         1     1  

[700 rows x 11 columns]

In [14]:
unscaled_inputs = data_with_targets.iloc[:,:-1]

**Standadize the data**

In [15]:
#from sklearn.preprocessing import StandardScaler

#absenteeism_scaler = StandardScaler()

In [16]:
from sklearn.base import BaseEstimator, TransformerMixin
from sklearn.preprocessing import StandardScaler

class CustomScaler(BaseEstimator, TransformerMixin):
    
    def __init__(self, columns, copy=True, with_mean=True, with_std=True):
        self.columns = columns
        self.copy = copy
        self.with_mean = with_mean
        self.with_std = with_std
        self.scaler = StandardScaler(copy=copy, with_mean=with_mean, with_std=with_std)
        self.mean_ = None
        self.var_ = None
    
    def fit(self, X, y=None):
        self.scaler.fit(X[self.columns], y)
        self.mean_ = np.mean(X[self.columns], axis=0)
        self.var_ = np.var(X[self.columns], axis=0)
        return self
    
    def transform(self, X, y=None, copy=None):
        init_col_order = X.columns
        X_scaled = pd.DataFrame(self.scaler.transform(X[self.columns]), columns=self.columns)
        X_not_scaled = X.loc[:, ~X.columns.isin(self.columns)]
        return pd.concat([X_not_scaled, X_scaled], axis=1)[init_col_order]


Our custom scaler will not standardize all inputs, but only the ones we choose

In [17]:
unscaled_inputs.columns.values

array(['Reason_1', 'Reason_2', 'Reason_3', 'Reason_4', 'Month Value',
       'Transportation Expense', 'Age', 'Body Mass Index', 'Education',
       'Children', 'Pets'], dtype=object)

In [18]:
#columns_to_scale = ['Month Value','Day of the Week', 'Transportation Expense', 'Distance to Work',
       #'Age', 'Daily Work Load Average', 'Body Mass Index', 'Children', 'Pets']
    
columns_to_omit = ['Reason_1', 'Reason_2', 'Reason_3', 'Reason_4','Education']

In [19]:
columns_to_scale = [x for x in unscaled_inputs.columns.values if x not in columns_to_omit]

In [20]:
absenteeism_scaler = CustomScaler(columns_to_scale)

In [21]:
absenteeism_scaler.fit(unscaled_inputs)

CustomScaler(columns=['Month Value', 'Transportation Expense', 'Age',
                      'Body Mass Index', 'Children', 'Pets'])

In [22]:
scaled_inputs = absenteeism_scaler.transform(unscaled_inputs)

In [23]:
scaled_inputs

Reason_1  Reason_2  Reason_3  Reason_4  Month Value  \
0           0         0         0         1     0.030796   
1           0         0         0         0     0.030796   
2           0         0         0         1     0.030796   
3           1         0         0         0     0.030796   
4           0         0         0         1     0.030796   
..        ...       ...       ...       ...          ...   
695         1         0         0         0    -0.568019   
696         1         0         0         0    -0.568019   
697         1         0         0         0    -0.568019   
698         0         0         0         1    -0.568019   
699         0         0         0         1    -0.568019   

     Transportation Expense       Age  Body Mass Index  Education  Children  \
0                  1.005844 -0.536062         0.767431          0  0.880469   
1                 -1.574681  2.130803         1.002633          0 -0.019280   
2                 -0.654143  0.248310         1.002633          0 -0.919030   
3                  0.854936  0.405184        -0.643782          0  0.880469   
4                  1.005844 -0.536062         0.767431          0  0.880469   
..                      ...       ...              ...        ...       ...   
695               -0.654143  0.562059        -1.114186          1  0.880469   
696                0.040034 -1.320435        -0.643782          0 -0.019280   
697                1.624567 -1.320435        -0.408580          1 -0.919030   
698                0.190942 -0.692937        -0.408580          1 -0.919030   
699                1.036026  0.562059        -0.408580          0 -0.019280   

         Pets  
0    0.268487  
1   -0.589690  
2   -0.589690  
3   -0.589690  
4    0.268487  
..        ...  
695 -0.589690  
696  1.126663  
697 -0.589690  
698 -0.589690  
699  0.268487  

[700 rows x 11 columns]

In [24]:
scaled_inputs.shape

(700, 11)

**Split of data into train and test and shuffle**

**Import the relevant module**

In [25]:
from sklearn.model_selection import train_test_split

**Split**

In [26]:
train_test_split(scaled_inputs, targets)

[     Reason_1  Reason_2  Reason_3  Reason_4  Month Value  \
 225         1         0         0         0    -0.268611   
 118         0         0         0         1     0.030796   
 616         0         0         0         1    -1.466241   
 611         1         0         0         0    -1.466241   
 164         1         0         0         0     0.030796   
 ..        ...       ...       ...       ...          ...   
 494         0         0         0         1     0.330204   
 22          1         0         0         0     0.929019   
 282         0         0         0         1     0.629611   
 600         1         0         0         0    -1.466241   
 386         0         0         0         1    -1.466241   
 
      Transportation Expense       Age  Body Mass Index  Education  Children  \
 225               -0.654143  0.248310         1.002633          0 -0.919030   
 118               -0.578689 -1.477309        -1.349389          0 -0.919030   
 616               -0.6541

In [27]:
x_train, x_test, y_train, y_test = train_test_split(scaled_inputs, targets, train_size= 0.8, random_state = 20)

In [28]:
print(x_train.shape,y_train.shape)

(560, 11) (560,)


In [29]:
print(x_test.shape,y_test.shape)

(140, 11) (140,)


The input contain 560 observations along 14 features, while the targets are a vector of length 560

### Logistic Regression with sklearn

In [30]:
from sklearn.linear_model import LogisticRegression
from sklearn import metrics

#### Training the model

In [31]:
reg = LogisticRegression()

In [32]:
reg.fit(x_train, y_train)

LogisticRegression()

In [33]:
reg.score(x_train, y_train)

0.7767857142857143

### Manually check the accuracy

In [34]:
model_outputs  = reg.predict(x_train)
model_outputs

array([0, 1, 1, 1, 1, 0, 0, 1, 0, 1, 1, 1, 0, 1, 0, 0, 0, 0, 0, 1, 1, 0,
       0, 0, 0, 0, 0, 1, 1, 0, 0, 1, 0, 1, 0, 0, 1, 0, 0, 0, 1, 0, 1, 1,
       0, 0, 0, 0, 1, 1, 0, 1, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0,
       0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 1, 0, 0, 1, 1, 1, 0, 0, 0, 1, 1, 0,
       0, 1, 0, 0, 1, 0, 0, 0, 0, 1, 0, 1, 1, 1, 1, 0, 1, 1, 0, 0, 1, 0,
       0, 1, 0, 1, 1, 0, 0, 0, 1, 1, 1, 1, 1, 0, 0, 1, 0, 1, 0, 0, 1, 0,
       0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 1, 0, 1, 0, 0, 1, 0,
       1, 1, 0, 1, 0, 1, 1, 0, 1, 0, 0, 1, 1, 0, 1, 1, 0, 0, 0, 0, 1, 0,
       0, 0, 0, 0, 1, 0, 0, 1, 1, 0, 0, 1, 0, 0, 1, 1, 1, 0, 0, 0, 0, 1,
       1, 0, 0, 1, 1, 1, 1, 0, 1, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 1,
       1, 0, 0, 0, 1, 0, 0, 1, 1, 0, 0, 1, 1, 0, 0, 1, 1, 0, 0, 0, 0, 1,
       0, 0, 1, 0, 1, 1, 1, 0, 1, 0, 1, 0, 1, 0, 1, 1, 0, 1, 1, 1, 1, 1,
       0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 1, 0, 1, 1, 0, 1, 0, 0, 1, 0, 1, 0,
       0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 0,

In [35]:
y_train

array([0, 1, 1, 1, 1, 0, 0, 1, 0, 1, 0, 1, 1, 0, 0, 0, 0, 0, 1, 1, 0, 0,
       1, 1, 0, 0, 0, 0, 1, 0, 0, 1, 0, 1, 0, 0, 0, 1, 1, 1, 1, 0, 1, 1,
       1, 0, 0, 0, 1, 1, 1, 1, 0, 0, 0, 1, 1, 0, 0, 1, 0, 0, 1, 0, 0, 0,
       0, 0, 0, 1, 0, 1, 1, 0, 0, 1, 1, 0, 0, 1, 1, 0, 0, 1, 0, 1, 0, 1,
       1, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 0, 1, 1, 0, 0, 1, 0,
       0, 1, 1, 1, 1, 0, 0, 1, 1, 1, 1, 1, 0, 0, 0, 1, 0, 1, 0, 0, 1, 1,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 1, 0, 0, 0, 1, 0, 0, 0, 1,
       0, 1, 0, 0, 0, 1, 1, 0, 1, 1, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 1, 0,
       0, 0, 0, 0, 1, 0, 0, 1, 1, 1, 0, 1, 0, 0, 0, 1, 1, 0, 0, 0, 0, 1,
       1, 0, 0, 1, 1, 1, 0, 0, 0, 1, 0, 0, 1, 1, 1, 0, 0, 1, 0, 1, 0, 0,
       1, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 1, 1, 1, 1, 0, 1, 1, 0, 0, 0, 0,
       0, 0, 1, 0, 1, 1, 1, 1, 1, 0, 1, 0, 1, 0, 1, 1, 1, 1, 1, 1, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1, 0, 1, 0, 1, 0, 0, 0, 1, 0, 0, 0,
       1, 0, 1, 0, 0, 0, 0, 1, 0, 1, 0, 0, 1, 1, 0,

In [36]:
model_outputs == y_train

array([ True,  True,  True,  True,  True,  True,  True,  True,  True,
        True, False,  True, False, False,  True,  True,  True,  True,
       False,  True, False,  True, False, False,  True,  True,  True,
       False,  True,  True,  True,  True,  True,  True,  True,  True,
       False, False, False, False,  True,  True,  True,  True, False,
        True,  True,  True,  True,  True, False,  True,  True,  True,
        True,  True,  True,  True,  True, False,  True,  True,  True,
        True,  True,  True,  True,  True,  True, False,  True,  True,
        True,  True,  True, False,  True,  True,  True,  True,  True,
       False,  True, False,  True,  True, False, False, False,  True,
        True,  True,  True,  True,  True,  True,  True, False,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True, False,  True,  True,  True,  True,
       False,  True,  True,  True,  True, False,  True,  True,  True,
        True,  True,

In [37]:
np.sum((model_outputs == y_train))

435

In [38]:
model_outputs.shape[0]

560

In [39]:
np.sum((model_outputs == y_train)) / model_outputs.shape[0]

0.7767857142857143

**Finding the Intercept and Coefficients**

In [40]:
reg.intercept_

array([-1.60957471])

In [41]:
reg.coef_

array([[ 2.77151176,  0.93168817,  3.09210221,  0.8090592 ,  0.00781237,
         0.62505482, -0.17390339,  0.28829409, -0.24081615,  0.35753531,
        -0.27337422]])

In [42]:
unscaled_inputs.columns.values

array(['Reason_1', 'Reason_2', 'Reason_3', 'Reason_4', 'Month Value',
       'Transportation Expense', 'Age', 'Body Mass Index', 'Education',
       'Children', 'Pets'], dtype=object)

In [43]:
feature_name = unscaled_inputs.columns.values

In [44]:
summary_table = pd.DataFrame(columns=['Feature name'], data = feature_name)

summary_table ['coefficient'] = np.transpose(reg.coef_)

summary_table

Feature name  coefficient
0                 Reason_1     2.771512
1                 Reason_2     0.931688
2                 Reason_3     3.092102
3                 Reason_4     0.809059
4              Month Value     0.007812
5   Transportation Expense     0.625055
6                      Age    -0.173903
7          Body Mass Index     0.288294
8                Education    -0.240816
9                 Children     0.357535
10                    Pets    -0.273374

In [45]:
summary_table.index = summary_table.index + 1

summary_table.loc[0] = ['Intercept', reg.intercept_[0]]

summary_table = summary_table.sort_index()
summary_table

Feature name  coefficient
0                Intercept    -1.609575
1                 Reason_1     2.771512
2                 Reason_2     0.931688
3                 Reason_3     3.092102
4                 Reason_4     0.809059
5              Month Value     0.007812
6   Transportation Expense     0.625055
7                      Age    -0.173903
8          Body Mass Index     0.288294
9                Education    -0.240816
10                Children     0.357535
11                    Pets    -0.273374

**Interpretation of the coefficients**

bias is also known as intercept and coefficient is also known as weight. The closer the weights are to zero, the smaller the weights and vise versa.

Whiever weight is bigger, its corresponding faeture is more important

Standardize coefficients are basically the coefficients of a regression where all variables have been standardized

In [46]:
summary_table ['Odds_ratio'] = np.exp(summary_table.coefficient)

In [47]:
summary_table

Feature name  coefficient  Odds_ratio
0                Intercept    -1.609575    0.199973
1                 Reason_1     2.771512   15.982778
2                 Reason_2     0.931688    2.538791
3                 Reason_3     3.092102   22.023327
4                 Reason_4     0.809059    2.245794
5              Month Value     0.007812    1.007843
6   Transportation Expense     0.625055    1.868348
7                      Age    -0.173903    0.840378
8          Body Mass Index     0.288294    1.334150
9                Education    -0.240816    0.785986
10                Children     0.357535    1.429801
11                    Pets    -0.273374    0.760808

In [48]:
summary_table.sort_values('Odds_ratio', ascending=False)

Feature name  coefficient  Odds_ratio
3                 Reason_3     3.092102   22.023327
1                 Reason_1     2.771512   15.982778
2                 Reason_2     0.931688    2.538791
4                 Reason_4     0.809059    2.245794
6   Transportation Expense     0.625055    1.868348
10                Children     0.357535    1.429801
8          Body Mass Index     0.288294    1.334150
5              Month Value     0.007812    1.007843
7                      Age    -0.173903    0.840378
9                Education    -0.240816    0.785986
11                    Pets    -0.273374    0.760808
0                Intercept    -1.609575    0.199973

NB: A feature is not particularly important:

-if its coefficient is around 0

-if its odds ratio is around 1

A weight(coefficient) of 0 implies that no matter the feature value, we will multiply it by 0 (in the model)

For a unit change in the standardized feature, the odds increase by a multiple equal to the odds ratio (1 = no change)

**Backward Elimination**

The idea of backward elimination is that we can simplify our model by removing all features which have close to no contribution to the model

when we have the p-values, we get rid of all coefficients with p-values >0.05

if the weight is small it won't make a difference anyway... if we romove these variables, the rest of our model should not really change in terms of coefficient values.

### Testing the model

In [49]:
reg.score(x_test, y_test)

0.7357142857142858

Based on the data the model has NEVER seen before, in 73.6% of the cases, the model will predict (correctly) if the person is going to be excessively absent.

Often time the test accuracy is 10-20% lower than the train accuracy (due to overfitting)

In [50]:
predicted_proba = reg.predict_proba(x_test)
predicted_proba

array([[0.75466347, 0.24533653],
       [0.60917591, 0.39082409],
       [0.48329093, 0.51670907],
       [0.75768232, 0.24231768],
       [0.08357741, 0.91642259],
       [0.3052464 , 0.6947536 ],
       [0.303675  , 0.696325  ],
       [0.11636888, 0.88363112],
       [0.7400284 , 0.2599716 ],
       [0.75596036, 0.24403964],
       [0.50609784, 0.49390216],
       [0.19501503, 0.80498497],
       [0.06248668, 0.93751332],
       [0.7055465 , 0.2944535 ],
       [0.29675526, 0.70324474],
       [0.52028649, 0.47971351],
       [0.50551315, 0.49448685],
       [0.50843643, 0.49156357],
       [0.36713074, 0.63286926],
       [0.06422143, 0.93577857],
       [0.73822433, 0.26177567],
       [0.75768232, 0.24231768],
       [0.47994423, 0.52005577],
       [0.47760936, 0.52239064],
       [0.22619725, 0.77380275],
       [0.74047815, 0.25952185],
       [0.51148533, 0.48851467],
       [0.87702735, 0.12297265],
       [0.24005377, 0.75994623],
       [0.75768232, 0.24231768],
       [0.

The two columns show probability of being 0 and 1 respectively

In [51]:
predicted_proba.shape

(140, 2)

In [52]:
predicted_proba[:,1]

array([0.24533653, 0.39082409, 0.51670907, 0.24231768, 0.91642259,
       0.6947536 , 0.696325  , 0.88363112, 0.2599716 , 0.24403964,
       0.49390216, 0.80498497, 0.93751332, 0.2944535 , 0.70324474,
       0.47971351, 0.49448685, 0.49156357, 0.63286926, 0.93577857,
       0.26177567, 0.24231768, 0.52005577, 0.52239064, 0.77380275,
       0.25952185, 0.48851467, 0.12297265, 0.75994623, 0.24231768,
       0.38859882, 0.71238179, 0.69821485, 0.49507156, 0.24231768,
       0.59772596, 0.26042186, 0.78022686, 0.4398591 , 0.60641126,
       0.24188848, 0.49713003, 0.25862385, 0.40689715, 0.80759194,
       0.59889345, 0.71944702, 0.24317762, 0.24676126, 0.2414598 ,
       0.50180816, 0.29299268, 0.6947536 , 0.24459387, 0.82033268,
       0.39193844, 0.90599346, 0.26442957, 0.32150234, 0.3220128 ,
       0.70502976, 0.69623972, 0.26579672, 0.77584327, 0.24541121,
       0.24490372, 0.07551158, 0.26087263, 0.76176924, 0.29640101,
       0.25772789, 0.31539593, 0.88408057, 0.4387068 , 0.59547

### Save the Model

In [53]:
import pickle

In [54]:
with open('model', 'wb') as file:
    pickle.dump(reg, file)

In [55]:
with open('scaler', 'wb') as file:
    pickle.dump(absenteeism_scaler, file)

NB: The information in the absenteeism_scaler is used to preprocess new data

In [58]:
file_path = r'C:\Users\MOJI\Desktop\PortfolioP\model'

# Save the model to the specified file path
with open(file_path, 'wb') as file:
    pickle.dump(reg, file)

In [59]:
file_path = r'C:\Users\MOJI\Desktop\PortfolioP\scaler'

# Save the model to the specified file path
with open(file_path, 'wb') as file:
    pickle.dump(absenteeism_scaler, file)